src/Microsoft.Azure.ServiceBus/Primitives/ITokenProvider.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.ServiceBus.Primitives
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Provides interface definition of a token provider.
+    /// </summary>
+    public interface ITokenProvider
+    {
+        /// <summary>
+        /// Gets a <see cref="SecurityToken"/>.
+        /// </summary>
+        /// <param name="appliesTo">The URI which the access token applies to</param>
+        /// <param name="timeout">The time span that specifies the timeout value for the message that gets the security token</param>
+        /// <returns><see cref="SecurityToken"/></returns>
+        Task<SecurityToken> GetTokenAsync(string appliesTo, TimeSpan timeout);
```

Comment:
>TimeSpan timeout) [](start = 60, length = 17)

Do we need this parameter? It is not used anywhere.

Reply:
we probably should use it

---
In reply to: [156552155](https://github.com/Azure/azure-service-bus-dotnet/pull/385#discussion_r156552155) [](ancestors = 156552155)

Reply:
This is required for someone building their own token provider and grabbing tokens from external services.

---
In reply to: [156552155](https://github.com/Azure/azure-service-bus-dotnet/pull/385#discussion_r156552155) [](ancestors = 156552155)

Reply:
Even if they build a token provider and grab a token from external service, do we support it in the service? We don't. I think we should remove this parameter as it is not used currently and most likely will not be used in the near future.